src/Datadog.Trace/Util/DomainMetadata.cs

```diff
@@ -113,9 +92,12 @@ private static void TrySetProcess()
         {
             try
             {
-                if (!_processDataPoisoned && _currentProcess == null)
+                if (!_processDataPoisoned && !_initialized)
                 {
-                    _currentProcess = Process.GetCurrentProcess();
+                    _initialized = true;
+                    _currentProcessName = ProcessHelpers.GetCurrentProcessName();
+                    _currentProcessMachineName = ProcessHelpers.GetCurrentProcessMachineName();
+                    _currentProcessId = ProcessHelpers.GetCurrentProcessId();
```

Comment:
This will create three `Process` instances that instantly become available for garbage collection. Should we combine these into a single call that returns the three values?

Reply:
I combined the three calls into one in 02a5a6aca09f19cf524c099aec8d1f58795467a9 . What do you think?

Reply:
Thanks!